### PR TITLE
Add support for defining `whiteLabelling.logoComponent` as a function

### PR DESCRIPTION
**Pull request by whage**
This change is about white labeling.
The template that generates the logo on the top of the page requires a
`React Element` and gets it from the `whiteLabelling.logoComponent` part of the global
configuration object.
Since `React` is not available in the scope of the global config object a React element
can't be created on the configuration side.
These changes make it possible for the `whiteLabelling.logoComponent` configuration parameter
to be a **function** which is passed the React object.
This way it is possible to call React.createElement() on the configuration
side and replace the logo with a custom element.

A possible usage:
```
window.config = {
  ...
  whiteLabelling: {
    logoComponent: function(React) {
      return React.createElement("img", {src: "https://www.sandraboynton.com/sboynton.com.data/Components/Photos/One-Shoe-Blues-DVD-menu.gif"})
    }
  }
}
```

A possible fix for https://github.com/OHIF/Viewers/issues/1407.

@dannyrb , can you please take a look?
I never worked with React before. If this is not the right way to do it then please give me some guidance and I'll try to make the right changes.

### CODE REVIEW

**Comment by dannyrb** [comment]
@whage, I admire your determination! Are/can React components return their type as `function`? If so, this gets a little hairy.

I think I would be okay with updating this to say that the user should supply a function that returns a React component, then we can _always_ pass `React`, and the consumer can use it if needed.

**Comment by whage** [comment]
@dannyrb , I don't know anything about react components, you tell me! What do you mean by "updating this to say..." ? Do you mean to document it somewhere? Or can we enforce logoComponenet to be a function somehow?

**Comment by NatiTM** [comment]
@dannyrb so the current implementation of whiteLabelling doesn't work?

**Comment by whage** [comment]
@NatiTM , it doesn't, because it needs a React element, but there is no way to create one on the configuration side (unless you already have react in your main app too). Maybe this should be renamed to be a bugfix?

**Comment by whage** [comment]
Why did this get assigned to me? I'm the one who opened it.

**Comment by NatiTM** [comment]
Usually this is how it is done on open source projects.
You can contribute.

**Comment by whage** [comment]
@NatiTM , What do you mean "you can contribute?" I'm the author of the code in this PR. We'd need *someone else* to review it I guess.

File: platform/viewer/src/connectedComponents/Viewer.js [hunk 1]
```diff
@@ -246,7 +246,7 @@ class Viewer extends Component {
             <UserManagerContext.Consumer>
               {userManager => (
                 <ConnectedHeader home={false} userManager={userManager}>
-                  {whiteLabelling.logoComponent}
+                  {(typeof whiteLabelling.logoComponent === 'function' ? whiteLabelling.logoComponent(React) : whiteLabelling.logoComponent)}
```
Comment (dannyrb) [inline platform/viewer/src/connectedComponents/Viewer.js]:
```suggestion
{{ whiteLabelling.createLogoComponentFn && whiteLabelling.createLogoComponentFn(React) }}
```

**Comment by dannyrb** [comment]
👋 @whage 
Sorry for the delay. When a 3rd party PR comes in, it enters a triage process and is prioritized. We only have funds for ~2 developers through our grant; white-labeling is technically outside the scope of the grant, so requests like these are addressed more slowly, and usually by me volunteering time, or one of our community partners sponsoring maintenance/development. 

I think we can update the configuration to add a new method under the whiteLabeling object:

`createLogoComponentFn(React) : ReactComponent`

It should take `React` as a parameter, and return a `ReactComponent` (or undefined/null)

So we need to:

- Update our config and docs
- Update the two files you've already touched

----

Long term, we would probably want to add a "Generic Logo Component" where you can provide an SVG, base64 encoded image, local image, text, link, etc. And then just use those as props for that component.

Then you have the ability to implement your own component, or simply configure the default one.

**Comment by arantes555** [comment]
This is not stale. It is still relevant.

**Comment by dannyrb** [comment]
Fixed Here: https://github.com/OHIF/Viewers/pull/1659

**Pull request closed without merging**